### PR TITLE
Implement request and respond for package version artifacts

### PR DESCRIPTION
**Pull request by tiainen**
## Description

Fixes pyrsia/pyrsia#623 pyrsia/pyrsia#624

This PR implements the following things:

 * requesting a package version artifact from the p2p network in the case where a docker manifest does not yet exist in the local pyrsia node
 * respond with the content of the package version artifact when another peer in the network requests it

## PR Checklist

- [x] I've read the [contributing guidelines](https://github.com/pyrsia/.github/blob/main/contributing.md).
- [x] I've read ["What is a Good PR?"](https://github.com/pyrsia/pyrsia/blob/main/docs/good_pr.md)
- [x] I've included a good title and brief description along with how to review them.
- [x] I've linked any associated an [issue](https://github.com/pyrsia/pyrsia/issues).
- [x] I've requested a review  from `pyrsia/collaborators`.

### Code Contributions

- [x] I've built the code `cargo build --all-targets` successfully.
- [x] I've run the unit tests `cargo test --workspace` and everything passes.
- [x] I've made sure my rust toolchain is current `rustup update`.

### CODE REVIEW

**Review by efrisch — APPROVED** [review]
lgtm. 👍

**Review by fishseabowl — APPROVED** [review]
LGTM 🚀

File: pyrsia_node/src/network/handlers.rs [hunk 1]
```diff
@@ -40,19 +42,63 @@ pub async fn provide_artifacts(mut p2p_client: Client) {
...
     channel: ResponseChannel<ArtifactResponse>,
 ) {
-    let decoded_hash = hex::decode(&hash.get(7..).unwrap()).unwrap();
-    match node_manager::handlers::get_artifact(&decoded_hash, HashAlgorithm::SHA256) {
+    debug!(
+        "Handling request artifact: {:?}={:?}",
+        artifact_type, artifact_hash
+    );
+    let content = match artifact_type {
+        ArtifactType::Artifact => get_artifact(artifact_hash),
```
Comment (chb0github) [inline pyrsia_node/src/network/handlers.rs:58]: Using an enum type called `ArtifactType` that isn't actually a type is very confusing. When I read artifact type I think `maven, gradle, cargo` etc

**Comment by chb0github** [comment]
A walk through of the code would be helpful. There is a lot here for me to absorb

File: pyrsia_node/src/network/handlers.rs (hunk 1 above)
Comment (tiainen) [reply pyrsia_node/src/network/handlers.rs:58]: In this context, an `Artifact` is anything that can be shared between two pyrsia nodes over the libp2p network. At the moment we can share:

 * a docker blob (managed as an artifact by the artifact manager)
 * a docker manifest (managed as a PackageVersion by the metadata manager)

I see the confusion that can arise between the artifacts shared between pyrsia nodes and the artifacts stored by the artifact manager. If necessary, we can come up with a better naming scheme.

**Pull request merged**